ipfs/bitswap/protocol.nim

```diff
@@ -0,0 +1,33 @@
+import pkg/chronos
+import pkg/libp2p/switch
+import pkg/libp2p/stream/connection
+import pkg/libp2p/protocols/protocol
+import ./stream
+
+export stream except readLoop
+
+const Codec = "/ipfs/bitswap/1.2.0"
+
+type
+  BitswapProtocol* = ref object of LPProtocol
+    connections: AsyncQueue[BitswapStream]
+
+proc new*(t: type BitswapProtocol): BitswapProtocol =
+  let connections = newAsyncQueue[BitswapStream](1)
+  proc handle(connection: Connection, proto: string) {.async.} =
+    let stream = BitswapStream.new(connection)
+    await connections.put(stream)
+    await stream.readLoop()
+  BitswapProtocol(connections: connections, codecs: @[Codec], handler: handle)
+
+proc dial*(switch: Switch,
```

Comment:
It's probably better to use peer event handler here similar to - https://github.com/status-im/nim-libp2p/blob/master/libp2p/protocols/pubsub/pubsub.nim#L538-L545

This will allow to automatically register/unretister a peer when it's connected.